src/crypto/hpke/mod.rs

```diff
@@ -0,0 +1,345 @@
+        let mut suite = EncryptionSuite::new(1, rng);
+        let match_key = new_share(1u64 << 39, 1u64 << 20);
+        let enc = suite.seal(0, match_key);
+
+        assert!(matches!(
+            suite.open(1, enc),
+            Err(DecryptionError::NoSuchKey(1))
+        ));
+    }
+
+    mod proptests {
+        use super::*;
+        use proptest::prelude::ProptestConfig;
+        use rand::Rng;
+
+        proptest::proptest! {
+            #![proptest_config(ProptestConfig::with_cases(50))]
+            #[test]
+            fn arbitrary_ct_corruption(bad_byte in 0..23_usize, bad_bit in 0..7_usize, seed: [u8; 32]) {
+                let rng = StdRng::from_seed(seed);
+                let mut suite = EncryptionSuite::new(1, rng);
+                let mut encryption = suite.seal(0, new_share(0, 0));
+
+                encryption.ct.as_mut()[bad_byte] ^= 1 << bad_bit;
+                let _ = suite.open(0, encryption).unwrap_err();
+            }
+        }
+
+        proptest::proptest! {
+            #![proptest_config(ProptestConfig::with_cases(50))]
+            #[test]
+            fn arbitrary_enc_corruption(bad_byte in 0..32_usize, bad_bit in 0..7_usize, seed: [u8; 32]) {
+                let rng = StdRng::from_seed(seed);
+                let mut suite = EncryptionSuite::new(1, rng);
+                let mut encryption = suite.seal(0, new_share(0, 0));
+
+                encryption.enc.as_mut()[bad_byte] ^= 1 << bad_bit;
+                let _ = suite.open(0, encryption).unwrap_err();
+            }
+        }
+
+        proptest::proptest! {
+            #![proptest_config(ProptestConfig::with_cases(50))]
+            #[test]
+            fn arbitrary_info_corruption(corrupted_info_field in 1..4, seed: [u8; 32]) {
+                let mut rng = StdRng::from_seed(seed);
+                let mut suite = EncryptionSuite::new(10, rng.clone());
+                let mut encryption = suite.seal(0, new_share(0, 0));
+
+                let mut site_origin = EncryptionSuite::<StdRng>::SITE_ORIGIN.to_owned();
+                let mut helper_origin = EncryptionSuite::<StdRng>::HELPER_ORIGIN.to_owned();
+
+                let info = match corrupted_info_field {
+                    1 => Info {
+                        key_id: encryption.info.key_id + 1,
+                        ..encryption.info
+                    },
+                    2 => Info {
+                        epoch: encryption.info.epoch + 1,
+                        ..encryption.info
+                    },
+                    3 => {
+                        let idx = rng.gen_range(0..site_origin.len());
+                        site_origin.remove(idx);
+
+                        Info {
+                            site_origin: site_origin.as_ref(),
+                            ..encryption.info
+                        }
+                    },
+                    4 => {
+                        let idx = rng.gen_range(0..helper_origin.len());
```

Comment:
You are removing a byte, which is probably OK, but there is also adding and modifying.  Could this use `origin in "[a-z]{1,10}"` or whatever proptest provides?

Reply:
fixed that as well